src/containers/habit-tracker/analytics/year-view/Heatmap.tsx

```diff
@@ -205,66 +174,40 @@ interface HeatmapProps {
   data: { name: string; data: { x: number; y: number; day: number }[] }[];
 }
 
+interface HeatmapChartProps {
+  options: ApexOptions;
+  data: HeatmapProps["data"];
+  loading: boolean;
+  isCompMounted: (isMounted: boolean) => void;
+}
+
 // Functional component to render the heatmap chart
 export function Heatmap({ data }: HeatmapProps) {
   const [loading, setLoading] = useState(true);
-  const [rendered, setRendered] = useState(false);
-  const [windowWidth, setWindowWidth] = useState(window.innerWidth);
-
-  // Event listener to update windowWidth on resize
-  useEffect(() => {
-    const handleResize = () => {
-      setWindowWidth(window.innerWidth);
-    };
-
-    window.addEventListener("resize", handleResize);
 
-    return () => {
-      window.removeEventListener("resize", handleResize);
-    };
-  }, []);
-
-  const handleRendered = () => {
-    console.log("Chart rendered, setting 'rendered' to true");
-    setRendered(true);
+  const isCompMounted = (isMounted: boolean) => {
```

Comment:
rename function to `isHeatmapChartMounted`.

Also, add a comment about why we are doing this extra step of checking the mounting. Explain the problem in brief in a comment.